Various optimizations and fixes

markh0338 opened the pull request:
Broke out argparse to its own function
Added ability to use text file with list of devices to gather l3 facts
A few other fixes

Commit:
Vaious optimizations and fixes

Broke out argparse to its own function
Added ability to use text file with list of devices to gather l3 facts
A few other fixes

austind commented on get_l3_facts.py:
```diff
@@ -6,6 +10,71 @@
+        "-f",
+        help="Text file with list of IPs and/or FQDNs (one per line) to query",
+        type=str,
+    )
```
Change option name to `--input` / `-i`

austind commented on get_l3_facts.py:
```diff
@@ -171,4 +196,4 @@ def main():
 
 if __name__ == "__main__":
-    main()
+    sys.exit(main(sys.argv))
```
What does this give us?

austind commented on get_l3_facts.py:
```diff
@@ -1,3 +1,7 @@
+#!/usr/bin/env python3
+
+import sys
+import os
```
Alphabetize imports

austind commented on get_l3_facts.py:
```diff
@@ -76,57 +144,9 @@ def save_csv(csv_path, output):
 
 
-def main():
+def main(argv):
```
No need to pass `argv`, `argparse` handles this.

austind commented on get_l3_facts.py:
```diff
@@ -6,6 +10,71 @@
 import napalm
 
 
+def get_args(args):
```
No need to take `args`, `argparse` handles it -- see comments below

austind commented on get_l3_facts.py:
```diff
@@ -76,57 +144,9 @@ def save_csv(csv_path, output):
-    if not args.secret:
-        args.secret = getpass.getpass("Enable secret: ")
-
+    args = get_args(argv[1:])
```
After removing `args` from `get_args()` and `main()`, call this as `get_args()`

austind reviewed (changes requested):
Looks good overall. See corrections. Thanks for contributing!

Commit:
requested changes made

markh0338 commented:
requested updates have been committed.

austind commented:
Alphabetize imports still pending

Commit:
Imports Work

Removed sys import
Reordered the rest of the imports and added space between standard library and 3rd party imports.

markh0338 commented:
Missed that. Reordered and separated 3rd party from standard libraries
